tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
```

Comment:
"predictable enough" makes me a bit nervous.  If you're confident the counts are predictable then it's fine, but there are definitely cases where there could be races.  E.g. if you write on one connection and listen on another, you may get an initial empty event first, or you may get the written data first, depending on which one hits the server first.